### ISSUE TITLE
Mistake in the focal loss implementation?

### ISSUE THREAD

**Issue by seva100** [ZhaoJ9014/face.evoLVe#180]
Hi, thank you very much for releasing this codebase -- it has been very useful to my project.

I'm wondering if there is a mistake in the Focal Loss implementation. The code in `loss/focal.py` first calculates CrossEntropy loss, **averages it over all samples**, and then applies a modulating factor `loss = (1 - p) ** self.gamma * logp`. If I understand the original [Focal Loss paper](https://arxiv.org/pdf/1708.02002.pdf) correctly, they propose to calculate CrossEntropy, apply the modulating factor, and **only then** average the result over all samples. 
I wonder, is the order changed on purpose in this repository? I think this way it might be losing the idea of Focal loss...

If it's actually a mistake, a simple fix in the line https://github.com/ZhaoJ9014/face.evoLVe/blob/63520924167efb9ef53dcceed0a15cf739cad1c9/loss/focal.py#L13

Other implementations also seem to be having a different order, e.g. see https://github.com/clcarwin/focal_loss_pytorch/blob/e11e75bad957aecf641db6998a1016204722c1bb/focalloss.py#L34

to `self.ce = nn.CrossEntropyLoss(reduction='none')` will suffice.
